src/pbrt/util/colorspace.cpp

```diff
@@ -3,19 +3,15 @@
 // SPDX: Apache-2.0
 
 #include <pbrt/util/colorspace.h>
-
-#ifdef PBRT_BUILD_GPU_RENDERER
-#include <pbrt/gpu/util.h>
-#endif
 #include <pbrt/options.h>
 
 namespace pbrt {
 
 #ifdef PBRT_BUILD_GPU_RENDERER
-PBRT_CONST RGBColorSpace *RGBColorSpace_sRGB;
-PBRT_CONST RGBColorSpace *RGBColorSpace_DCI_P3;
-PBRT_CONST RGBColorSpace *RGBColorSpace_Rec2020;
-PBRT_CONST RGBColorSpace *RGBColorSpace_ACES2065_1;
+PBRT_GPU __constant__ RGBColorSpace *RGBColorSpace_sRGB;
```

Comment:
The intent with `PBRT_CONST` defined in `pbrt.h` is to handle the messiness of device-side constant buffers. Does that not work here? In general I'd rather keep using `PBRT_CONST` but potentially #define it differently for AMD targets in `pbrt.h`. (Related, does changing it to effectively `__device__ __constant__` as done here work on NVIDIA? I assume so but we should be sure.)

Reply:
`PBRT_CONST` is defined as `__device__ const` in `pbrt.h` not `__constant__`. I need to check the actual error message, but I remember HIP was complaining in `hipMemcpyToSymbol` and `hipMemcpyFromSymbol` functions.